### PR TITLE
TP-667: Add javadoc to converters and deprecate converters present in Spring

### PR DESCRIPTION
**Pull request by Bjornelmers**
TP-667: Add javadoc to custom converters and deprecate converters already present in Spring

* Add test cases to clarify which converters can be deprecated and not.
* Currency converters are currently the only custom converters that we provide that have identical behavior as the ones in Spring.
* The Instant- and LocalDateTime converters lose precision when relying on the standard Spring converters. The reason is that it is saved as a Date and not a String.
* The LocalDate-, YearMonth and TimeLocalDateTime converters give the same end result when first writing and then reading, however the object type that is written differs from standard Spring.
* Minor fixes in README.md.

### CODE REVIEW

File: ymer/src/main/java/com/avanza/ymer/support/JavaLocalDateTimeReadConverter.java [hunk 1]
```diff
@@ -15,12 +15,17 @@
...
-
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.data.convert.ReadingConverter;
+
+/**
+ * Converter that converts from a String to a {@link LocalDateTime}. This is a good choice if high time precision
+ * is more important that saving space. Otherwise {@link JavaTimeLocalDateTimeReadConverter} could be
```
Comment (kbs4j) [inline ymer/src/main/java/com/avanza/ymer/support/JavaLocalDateTimeReadConverter.java:26]: Typo? "is more important _**than**_ saving space"

File: ymer/src/main/java/com/avanza/ymer/support/JavaLocalDateTimeWriteConverter.java [hunk 2]
```diff
@@ -15,12 +15,17 @@
...
-
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.data.convert.WritingConverter;
+
+/**
+ * Converter that converts from a {@link LocalDateTime} to a String. This is a good choice if high time precision
+ * is more important that saving space. Otherwise {@link JavaTimeLocalDateTimeWriteConverter} could be
```
Comment (kbs4j) [inline ymer/src/main/java/com/avanza/ymer/support/JavaLocalDateTimeWriteConverter.java:26]: Typo?

File: ymer/src/main/java/com/avanza/ymer/support/JavaTimeLocalDateTimeReadConverter.java [hunk 3]
```diff
@@ -15,13 +15,18 @@
...
-
 import java.time.Instant;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
 
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.data.convert.ReadingConverter;
+
+/**
+ * Converter that converts from a Long to a {@link LocalDateTime}. This is a good choice if
```
Comment (kbs4j) [inline ymer/src/main/java/com/avanza/ymer/support/JavaTimeLocalDateTimeReadConverter.java:26]: Perhaps Long should have @link as well? To be consistent

File: ymer/src/test/java/com/avanza/ymer/ConverterTest.java [hunk 4]
```diff
@@ -115,21 +330,65 @@ public void setFile(String file) {
...
+			return yearMonth;
+		}
+
+		public void setYearMonth(YearMonth yearMonth) {
+			this.yearMonth = yearMonth;
+		}
+
+		public LocalDateTime getLocalDateTimeMillis() {
+			return localDateTimeMillis;
+		}
```
Comment (kbs4j) [inline ymer/src/test/java/com/avanza/ymer/ConverterTest.java:388]: Great that you extended the tests so thoroughly! ⭐ 👍

File: ymer/src/main/java/com/avanza/ymer/support/JavaLocalDateTimeReadConverter.java (hunk 1 above)
Comment (Bjornelmers) [reply ymer/src/main/java/com/avanza/ymer/support/JavaLocalDateTimeReadConverter.java]: Good catch! Fixed in a separate commit.

File: ymer/src/main/java/com/avanza/ymer/support/JavaLocalDateTimeWriteConverter.java (hunk 2 above)
Comment (Bjornelmers) [reply ymer/src/main/java/com/avanza/ymer/support/JavaLocalDateTimeWriteConverter.java]: Good catch! Fixed in a separate commit.

File: ymer/src/main/java/com/avanza/ymer/support/JavaTimeLocalDateTimeReadConverter.java (hunk 3 above)
Comment (Bjornelmers) [reply ymer/src/main/java/com/avanza/ymer/support/JavaTimeLocalDateTimeReadConverter.java]: Sure! Why not? Have added links to all the converter-classes in a separate commit.

File: ymer/src/test/java/com/avanza/ymer/ConverterTest.java (hunk 4 above)
Comment (Bjornelmers) [reply ymer/src/test/java/com/avanza/ymer/ConverterTest.java:388]: Thanks! It was necessary to really understand which converters that could be deprecated and not.

File: ymer/src/main/java/com/avanza/ymer/support/CurrencyWriteConverter.java [hunk 5]
```diff
@@ -15,11 +15,20 @@
...
 import org.springframework.data.convert.WritingConverter;
 
-import java.util.Currency;
-
+/**
+ * Converter that converts from a {@link Currency} to a {@link String}.
+ *
+ * @deprecated
+ * The converter {@code CurrencyToStringConverter}
+ * already exist in Spring ({@code MongoConverters}).
```
Comment (pativa) [inline ymer/src/main/java/com/avanza/ymer/support/CurrencyWriteConverter.java:28]: Maybe these could be @link to these classes?

File: ymer/src/test/java/com/avanza/ymer/ConverterTest.java (hunk 4 above)
Comment (pativa) [reply ymer/src/test/java/com/avanza/ymer/ConverterTest.java:388]: I agree, nice with good test coverage ⭐

**Review by pativa — APPROVED** [review]
Looks good! 👍

**Review by ath0s — APPROVED** [review]
Bonus ⭐ for the tests! 👍

File: ymer/src/main/java/com/avanza/ymer/support/CurrencyWriteConverter.java (hunk 5 above)
Comment (pativa) [reply ymer/src/main/java/com/avanza/ymer/support/CurrencyWriteConverter.java:28]: You're right, javadoc does not support linking to package-private classes (as they're not supposed to be accessed)... So I guess using @code is the best way to refer to this classes, just as you did 👍

**Pull request merged**